packages/xo-server-backup-reports/src/index.js

```diff
@@ -146,72 +221,162 @@ class BackupReportsXoPlugin {
   }
 
   test({ runId }) {
-    return this._backupNgListener(undefined, undefined, undefined, runId)
+    return this._report(runId, undefined, true)
   }
 
   unload() {
     this._xo.removeListener('job:terminated', this._report)
   }
 
-  _wrapper(status, job, schedule, runJobId) {
-    if (job.type === 'metadataBackup') {
-      return
-    }
+  async _report(runJobId, { type, status } = {}, force) {
+    const xo = this._xo
+    try {
+      if (type === 'call') {
+        return this._legacyVmHandler(status)
+      }
 
-    return new Promise(resolve =>
-      resolve(
-        job.type === 'backup'
-          ? this._backupNgListener(status, job, schedule, runJobId)
-          : this._listener(status, job, schedule, runJobId)
-      )
-    ).catch(logError)
+      const log = await xo.getBackupNgLogs(runJobId)
+      if (log === undefined) {
+        throw new Error(`no log found with runId=${JSON.stringify(runJobId)}`)
+      }
+
+      const reportWhen = log.data.reportWhen
+      if (
+        !force &&
+        (reportWhen === 'never' ||
+          (reportWhen === 'failure' && log.status === 'success'))
+      ) {
+        return
+      }
+
+      const job = await xo.getJob(log.jobId)
+      const schedule = await xo.getSchedule(log.scheduleId).catch(error => {
+        logger.warn(error)
+      })
+
+      return job.type === 'backup'
+        ? this._ngVmHandler(log, job, schedule, force)
+        : this._metadataHandler(log, job, schedule, force)
```

Comment:
Explicitly check the type here to avoid an unknown type to be handled as a metadata backup.